File: Models/Post.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace turism.Models
{
    public class Post
    {
      
        public int Id { get; set; }
        public string Type { get; set; }
        public string Title {get; set;}
        public string PostText {get; set;} 
        public string GetThere { get; set; }
        public int Approved { get; set; }

        public DateTime DateAdded {get; set;}
        // [JsonIgnore]
        public ICollection<Photo> Photos {get;set;}
        
        public User User {get; set;}

        public int UserId { get; set; }

        public City City {get; set;}

        public int CityId {get; set;}

        public ICollection<PostLike> PostLikes {get;set;}

        public ICollection<Reply> Replies {get; set;}

        public static implicit operator List<object>(Post v)
        {
            throw new NotImplementedException();
        }
    }
}

File: Turism-SPA/src/app/.model/post.ts
import { Photo } from './photo';
import { Reply } from './reply';
import { Like } from './like';
import { User } from './user';
import { City } from './city';

export interface Post {
    id: number;
    postText: string;
    approved: number;
    title: string;
    type?: string;
    getThere?: string;
    dateAdded: Date;
    userId: number;
    photos?: Photo[];
    cityId: number;
    postLikes?: Like[];
    replies?: Reply[];
    user: User;
    city: City;

}
